### ISSUE TITLE
New version handling transition's issues

### ISSUE THREAD

**Issue by theevann** [theevann/DD3#11]
Here are the code issues we can work on :    
- [x] Problem of modification of a shape (from a concurrent transition or an attribute change) during a transition - [Need to recompute probable recipients taking in account all transitions]
- [x] Problem of starting to watch an element during a transition - [Should we support this ? it is possible to support it, though it will again be very hacky]
- [x] Problem of interrupt - [Not handled]
- [x] Problem of subtransitions - [Partly handled as they trigger start - we don't get a chance to remove previous recipients as it doesn't emit end event... need to dig into this issue]
- [x] Problem of tween function (attrTween, styleTween, tween). As we don't send functions, the tween functions have to exist and be clearly identified on each browser - [Not handled]
- [x] Transitions don't handle correctly transform attribute
- [x] Un-homogeneous order of data in receivers dom between browsers
- [x] Groups problem. For now, groups are not handled at all. If a group is watched, we should watch the group and its children for changes (I think the children's attribute **watch** should be set to true in order not to have to go through all the dom tree each time we send something). Should we send the group itself ? Probably, and with all what it contains. 
- [x] Groups transitions ... I still need to think, but it might be as simple as creating a transition for the group and send it.

**Comment by theevann** [comment]
I propose not to support to start to watch an element during a transition - It's feasable but it's better if users take the habit of watching object before doing anything with it.

Also, what do you think we should do when we want to "unwatch" an object ? Should we delete it from all current recipients ?

**Comment by theevann** [comment]
I made so that elements received are ordered according to an absolute order depending on the sender and the id. We may in the future provide functions to specify which element to put in which order

**Comment by shoringfan10** [comment]
'Users take the habit of watching object before doing anything with it', do you mean that users, at this stage, might need to wait the transition to be done before doing the next?

If an 'unwatched' object will be re-watched by the same recipients from its very last state, maybe we could also set a flag to de-activate it in recipients.

**Comment by theevann** [comment]
No, actually I meant that the case may happen that for example a user starts a transition on an object not watched and then decide to send this object to peers. We would then have to find the current started transitions on this object, estimate recipients and send it. It's possible but it is better to "watch" the object before sending it.

However, I just changed the default behavior so that every objects is by default watched for changes and if needed sent. It's more logical as the user don't have to specify for each object that he wants it to be watched for changes (It's what I would expect). We can still request not to watch an object, though this should happen really rarely. (For axis for example)

I just made some important changes to the way elements are displayed. It might still be unstable.
